test/sparsematrix_ops.jl

```diff
@@ -491,4 +491,21 @@ end
     f()
     @test f() == 0
 end
+
+@testset "Comparisons to adjoints are efficient" for
+    A in Any[sparse(I(10000)), sprandn(10000, 10000, 0.00001), sprandn(ComplexF64, 100, 100, 0.9)],
+    B in Any[sparse(I(10000)), sprandn(10000, 10000, 0.00001), sprandn(ComplexF64, 100, 100, 0.9)]
+    if size(A) == size(B)
+        # Run each test twice to ignore compile times
+        t1 = @elapsed A == B
+        t1 = @elapsed A == B
+        As = Any[A, A', transpose(A), transpose(A'), transpose(A)']
+        Bs = Any[B, B', transpose(B), transpose(B'), transpose(B)']
+        t2 = maximum([(@elapsed A′ == B′) for A′ in As, B′ in Bs])
+        t2 = maximum([(@elapsed A′ == B′) for A′ in As, B′ in Bs])
+        # If it's inefficient, it's really so -- the O(n²) algorithm will stick out
+        @test t2 ≤ 100*t1
```

Comment:
this might be a bad idea, CI is not necessarily have capacity

Reply:
How big are these matrices?

Reply:
what i mean that it might be slow for what ever reason. i.e. capacity to do this fast enough.

Reply:
The current test code has a bug, it also times the dense case.

The matrices have about 10k nonzero entries. Some are sparse (10k^2), others dense.

Yes, I'm unsure a timing test is a good idea. The alternative is to define a new datatype that counts comparison operations.

Reply:
Timing doesn't work. Will need a new datatype.